### ISSUE TITLE
pretty-printing of closed poly-variant types seems broken in one case

### ISSUE THREAD

**Issue by chetmurthy** [ocaml/ocaml#9548]
Version: 4.10.0, also found same code in 4.11.0-alpha1

The pretty-printing of a closed variant type
```
type t = [ | w ]
```
produces
```
type t = [ w]
````
but this is not itself parseable:
```
# type t = [ | w ] ;;
type t = w
# type t = [ w ] ;;
Line 1, characters 13-14:
1 | type t = [ w ] ;;
                 ^
Error: Syntax error
```
A type with a variant and not inheritance as the first branch (a "row_list" and not a "core_type") is parseable either with or without the "|":
```
# type w = [ `A of int ];;
type w = [ `A of int ]
# type w = [ | `A of int | z ];;
type w = [ `A of int | `B of int ]
```
Not sure how important this is, though it would seem that ensuring that the pretty-printer always emits parseable output would be valuable.

I'm not sure how this should be fixed, but with some feedback, I could provide a fix.  One could either:
(a) always emit the leading "|" (which I think is safe, but not sure)
or (b) check that the first field is a row_list, and not a core_type (and if not, then emit the "|").

If someone could suggest which is the right course, I can code a fix against 4.11.0-alpha1.

**Comment by Octachron** [comment]
Option b seems more æsthetically pleasing  to me.

**Comment by Octachron** [comment]
Fixed in #9591 .
